### PR TITLE
feat: use `pnpm` instead of `yarn` berry

### PR DESCRIPTION
**Pull request by jbroma**
### Summary

Why pnpm? https://pnpm.io/motivation

#### Pros:
- dependency installation speed
- great deduplication of dependencies -> much less disk space used
- stricter scope for dependencies

#### Cons:
- Podfile needs to use custom script to resolve location of CLI dynamically as it is inside virtual-store (this has already landed in RN 0.72)
- Gradle scripts need to use similar approach to resolve the location of CLI dynamically
- Re.Pack output is a little bit harder to read as some paths now refer to `pnpm`'s virtual store instead of nearest `node_modules` (potentially fixable but shouldn't be an issue).

### Checklist:
- [x] - fix typecheck
- [x] - fix lint
- [x] - fix tests
- [x] - replace `yarn` with `pnpm` in docs
- [x] - fix lint setup in remaining packages
- [x] - add helper scripts for gradle & pods 
- [x] - make mocks more reusable within monorepo
- [x] - add pnpm version to `package.json` in root of the monorepo
- [x] - add section in docs about pnpm - why it's used, how to install it (link) etc. 

### Benchmarks

| Tool       | real time | user time | sys time | Disk Space |
|------------|-----------|----------|----------|------------|
| yarn berry | 412.93s   | 24.96s   | 102.67s  | 3577MB     |
| yarn berry (pnpm) | 11.5s | 8.8s | 14.07s | 578MB | 
| pnpm       | 7.34s     | 4.37s    | 16.23s   | 528MB      |

#### Notes:
- results after `git clone` and installing dependencies
- for `yarn` the installation phase is very quick, but the linking phase takes 95% of it's time
- `yarn` is using `hoisitingLimits: workspaces`
- `pnpm` is using the default setting for `node-linker: isolated`
- sys time greater than real time indicates using multiple threads to finish the operation (`pnpm` uses  `--workspace-concurrency=4` by default)

### CODE REVIEW

File: packages/host/jest.setup.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            throw new Error(
+              `jest.setup.js: unknown module: ${module} from container: ${container}`,
+            );
+        }
+      }
+    }),
+  },
+}));
+
+jest.useFakeTimers();
```
Comment (thymikee) [inline packages/host/jest.setup.js:88]: if we run it in the setup, maybe let's set this in jest's config file?

**Review by thymikee — APPROVED** [review]
damn that's faster

File: packages/host/jest.setup.js (hunk 1 above)
Comment (jbroma) [reply packages/host/jest.setup.js:88]: oh nice, didn't know jest had an option for global `fakeTimers`, will fix!

File: README.md [hunk 2]
```diff
@@ -62,92 +62,94 @@ Each of the mini apps could be deployed and run as a standalone app.
 
 ## How to use
 
+This project now uses `pnpm` instead of `yarn`. Learn how to install `pnpm` [here](https://pnpm.io/installation)
```
Comment (thymikee) [inline README.md:65]:
```suggestion
We use `pnpm` to manage dependencies. Learn how to install `pnpm` [here](https://pnpm.io/installation)
```

File: packages/auth/package.json [hunk 3]
```diff
@@ -2,30 +2,27 @@
   "name": "auth",
   "version": "0.0.1",
   "private": true,
-  "installConfig": {
-    "hoistingLimits": "workspaces"
-  },
   "scripts": {
     "start": "react-native webpack-start --port 9003",
-    "test": "jest",
+    "test": "jest --passWithNoTests",
```
Comment (thymikee) [inline packages/auth/package.json:7]: leftover?
```suggestion
    "test": "jest",
```

**Review by thymikee — APPROVED** [review]

File: packages/auth/package.json (hunk 3 above)
Comment (jbroma) [reply packages/auth/package.json:7]: no tests there in auth, so that's required

**Pull request merged**